pkg/appstatus/client.go

```diff
@@ -50,28 +50,41 @@ func (c *Client) WaitForAppDeployment(ctx context.Context, appName, orgNamespace
 	defer ticker.Stop()
 
 	ctx, cancel := context.WithTimeout(ctx, timeout)
+	deadline := time.Now().Add(timeout)
 
 	go func(appName, orgNamespace string) {
-		// Get app status using kubectl
-		kubectlCmd := exec.Command("kubectl", "get", "app", appName,
-			"-n", orgNamespace,
-			"-o", "jsonpath={.status.release.status}")
-		kubectlCmd.Stderr = c.stderr
-
-		output, err := kubectlCmd.Output()
-		if err == nil {
-			status := string(output)
-			if status == "deployed" {
-				c.logger.Infof("App %s successfully deployed", appName)
+		for {
+			if time.Now().After(deadline) {
+				c.logger.Errorf("App %s was not deployed within %v", appName, timeout)
```

Comment:
You don't need this as the timeout is already in the context. But we could add a check for ctx.Done to ensure the goroutine is actually stopped when timeout is reached.

Somethine like


```go
	select {
  		case <-ctx.Done():
  			return ctx.Err()
  		case out <- ticker.C:
			// nothing here, let the function run.
  	}
```